### ISSUE TITLE
'nan_policy' variable getting overridden in stats.compare_images()

### ISSUE THREAD

**Issue by VictorBarnes** [netneurolab/neuromaps#77]
### Description of issue

Hi neuromaps team,

When running stats.compare_images(), if nan_policy is set to 'omit' and nulls is set to None then nan_policy will be overridden to 'propagate' (the default value for efficient_pearsonr()) when the metric function is called. The reason for this seems to be because the nan_policy variable is not passed into the metric function call (line 89 of stats.py), whereas the nan_policy variable is passed into the permtest_metric() function (line 86-87 of stats.py).

A quick solution would be to change line 89 of stats.py to the following:
`return metric(srcdata, trgdata, nan_policy=nan_policy)`

However, this change may cause follow-up issues if the metric used is not one of the defaults ('pearsonr' or 'spearmanr') because the (non-default) metric function might not recognize the nan_policy variable. 

Thanks in advance,
Victor

### Code of Conduct

- [X] I agree to follow the `neuromaps` Code of Conduct

**Comment by VinceBaz** [comment]
Hey @VictorBarnes, sorry for the late reply! Thank you for pointing out this issue :smile: 

I took a look at the code, and I think its ok that `nan_policy` is overridden to `propagate` when `efficient_pearsonr` is called: we expect `compare_images` to handle this `nan_policy` first and remove the `nan` values when it's set to `omit`. When we call `efficient_pearsonr`, it then doesn't matter what `nan_policy` is set to because we've removed the `nan` values from the data arrays. 

I did find a different bug, however, which I fixed:  `nan` values are dropped from `srcdata` and `trgdata` regardless of whether `nan_policy` is set to `omit` or `propagate`. This is not what we want. We want to keep the `nan` values when `nan_policy` is set to `propagate`, so that the metric function can handle these `nan` values however it wants. Maybe this was related to your issue?

Let me know if it all makes sense, and sorry again for the late reply!
